datadog/resource_datadog_role.go

```diff
@@ -36,6 +41,11 @@ func resourceDatadogRole() *schema.Resource {
 					Required:    true,
 					Description: "Name of the role.",
 				},
+				"default_permissions_opt_out": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Description: "If set to `true`, the role does not have default (restricted) permissions unless they are explicitly set. The `include_restricted` attribute for the `datadog_permissions` data source must be set to `true` to manage default permissions in Terraform.",
```

Comment:
> The `include_restricted` attribute for the `datadog_permissions` data source must be set to `true` to manage default permissions in Terraform.

interesting. it seems to me like we might want to make the datadog permissions datasource always include restricted permissions now. is there any reason to omit them?

Reply:
agree, this seems like a backwards-compatible change so it's probably not a major risk?

Reply:
yeah if we can gain confidence it's safe to make that change i think it would be preferable as it simplifies the interface

Reply:
from discussion in slack:
> changing the data source to expose restricted permissions by default is riskier to do, especially if not all customers have access to this outright

since we're currently in a state where an FF needs to be enabled for these changes to work, we probably shouldn't expose all the permissions _yet_ since it might lead them to confusion when a role gets created w/ restricted perms anyways